From a57df696280994a6903025dde4694c8d6590bad5 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Sat, 21 Jan 2023 19:14:39 -0800
Subject: [PATCH] Set core options in retro_set_environment()

---
 libretro/libretro-core.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libretro/libretro-core.c b/libretro/libretro-core.c
index bcf2279ad..04f4fcd09 100644
--- a/libretro/libretro-core.c
+++ b/libretro/libretro-core.c
@@ -5179,6 +5179,8 @@ void retro_set_environment(retro_environment_t cb)
    if (environ_cb(RETRO_ENVIRONMENT_GET_VFS_INTERFACE, &vfs_iface_info))
       filestream_vfs_init(&vfs_iface_info);
 #endif
+
+   retro_set_core_options();
 }
 
 int log_resources_set_int(const char *name, int value)
@@ -7509,7 +7511,6 @@ void retro_init(void)
 
    /* Must set these here for the dynamic cartridge option */
    retro_set_paths();
-   retro_set_core_options();
 
    /* Clean ZIP temp */
    if (!string_is_empty(retro_temp_directory) && path_is_directory(retro_temp_directory))
-- 
2.34.1

